### DISCUSSION TITLE
RTKQ selectfromresult equailty checking

### DISCUSSION THREAD

**Discussion by niZmosis** [reduxjs/redux-toolkit#1465] [Q&A]
I recently reworked my code so I am no longer keeping my own state in the store, trying to make it stay in sync with RTKQ. What I liked about it was being able to use useSelector which has an equaity check. I am now using selectfromresult to call the query cache. The problem is I am polling the actual query every 10 seconds (its a game) and the selectfromresult is firing everytime even though it has the same results. Is there a way i can use something like lodash isEqual with selectfromresult to prevent rerenders? Is there a different method to prevent rerenders? Thanks! Oh and RTK is awesome!

**Comment by markerikson** [comment]
Can you show what your code looks like?

`selectionFromResult` _should_ be doing a shallow equality check, which means that the fields being returned need to be stable references from run to run:

https://redux.js.org/tutorials/essentials/part-8-rtk-query-advanced#selecting-values-from-results

**Reply by niZmosis** [reply]
I have noticed in other places that it does shallow checking. In this particular instance, its an array with strings. Here is the query, then the selectrfromresults.

`const { data: openLubAuctions } = useGetUserLubAuctionsOpenQuery(
        { address: account },
        {
            skip: !account || authLevel < 2,
            pollingInterval: pollingIntervalUserAuctions
        }
    )`

`const { previousPlacedBids } = userApi.endpoints.getUserLubAuctionsOpen.useQueryState(
        { address: account },
        {
            selectFromResult: ({ data }) => ({
                previousPlacedBids:
                    data?.entities?.[auctionId]?.bidsArray?.map((bid) =>
                        fixNumberString(bid.bidNumber / 100)
                    ) ?? []
            })
        }
    )`

Im going to read over that link you sent now.

Ohh I see now. So I should create a custom memoized selector that I use with the useQueryState. I dunno how many times I've read these docs, i don't remember that one lol.

**Reply by markerikson** [reply]
Yeah, it's the `.map()` call that's the issue.

`lastResult.previousPlacedBids !== currentResult.previousPlacedBids` because they're different arrays, which is the same scenario that the link actually happens to show for why we need a memoized selector here.

**Reply by niZmosis** [reply]
Awesome, thank you.
